### ISSUE TITLE
Important. Read it.

### ISSUE THREAD

**Issue by FallenBO** [Budschie/BudschieMorphMod#91]
Hello, I have a couple of questions for you. Why instead of making the mod more usable, with more features, do you leave it raw and just port it to a higher version? Why not fixate, for example, on version 1.18.2 and finalize it? Take for example the same metamorph mod, the menu is much more convenient. Can I contact you via Discord, it will be more convenient for both of us, I think. I can point out all the flaws. Super mod, but very damp.

**Comment by Budschie** [comment]
Well, I gotta admit that the morph UI is a bit dumb to navigate at times. Anyways, I'd discuss the flaws of this mod here in the GitHub issues rather than in my Discord DMs, as this would integrate better into my workflow and it would also be more interesting if other people could take part in this conversation too.
\- Budschie

**Comment by FallenBO** [comment]
Thanks for the answer. In that case, here are the questions:
When will the sounds and animations be available? In terms of sounds, I think it's clear what I mean, but about animation. Most aggressive mobs strike with their left hand. An iron golem, for example, basically has no hit animation, and throws up too high when compared to a regular mob. Nether mobs don't have fire resistance, Piglins attack whoever you are. The skeleton doesn't burn, and the zombie doesn't have 2 armor.
Of course, I don’t understand how to write mods, but now in minecraft there is an “attribute” command (adjusts the characteristics), and the Blaze has too fast projectiles.
There is actually a lot to discuss.
I don't know what the word limit is here, but if needed, I can try to write the whole list of problems.

**Comment by Budschie** [comment]
Yeah, those are oversights made by me.

**Comment by Budschie** [comment]
I especially agree with the animation stuff, I should probably reconsider how I handle animations in the current mod. As for the sounds, I could probably implement them, though I would find them a bit annoying to be honest. The +2 armor stuff is another oversight by me, but fixing this shouldn't pose too big of a challenge, I would probably only have to add one JSON file to my mod. The behaviour of piglins is also something I noticed; I should probably fix it.

The only two things I don't necessarily agree with are the speed of the blaze fireballs, they are equal to the vanilla blaze projectile speed, and the whole skeletons don't burn in fire thing (because skeletons actually burn in fire, I tested it).

**Comment by Budschie** [comment]
The zombies have armor points now thanks to this commit: a100cf1a8dfbba6fc0e43ae9d550af3d0be916c2

**Comment by FallenBO** [comment]
Tomorrow I will write you a list of problems and suggestions to make it easier for you to fix it.
By the way, I don’t understand a little, does the fix happen immediately for all versions of Minecraft or not?

**Comment by FallenBO** [comment]
**List of bugs** _and_ **shortcomings I found.** _Minecraft_ _version_ 1.17.1

- 

1. Enderman, Vex, Piglin/Brutal or any other mob from any mod - Strike visually with the left hand, not the right.
2. Hoglin, Piglin/Brutal, in the Overworld, after 15 seconds, do not turn into a zombified version of themselves.
3. Blaze, Enderman - have a long cooldown ability (Compared to Vanilla mobs).
4. Slime, Magma Cube, (Guardian, Ancient Guardian, Fish on land) - do not have a jump animation when walking.
5. Magma Cube - does not have a unit of armor (Small Cube - 3, Medium Cube - 6, Large Cube - 12), fire / lava protection (as well as Strider, Wither Skeleton, Zombie Piglin). Has no choice of size (Slime too), has no jump height (Small Cube - 1 block, Medium - 1.5 blocks, Large - 3 blocks), takes fall damage.
6. Spiders have high speed and wall climb (As if the player were flying in creative). One block is like a half block for them. Do not take fall damage.
7. Turning into Skeletons, Zombified mobs - the player drowns.
8. Strider does not take damage in water/rain and does not walk on lava. Does not turn pale when not in lava.
9. If a player, being a Zombie, infected a villager, it happens that this Zombie villager starts attacking the player.
10. If you hit an aggressive mob, being one of them yourself - mobs nearby will start attacking the player.
11. Villager do not run away from the player if he turned into a Zombie.
12. As a Villager, aggressive mobs will start attacking you (only Zombies and (Zoglin) should).
13. As a golem, aggressive mobs will start attacking you even if you don't hit them (should only Spiders, Zombie, Skeletons and (Zoglin)).
**14**. Golem throws too high and has no hit animation (Compared to Vanilla mob).
15. Being an aggressive mob, golem and snowman (defender mobs) won't attack you (they won't attack only creepers).
16. Piglin/Brutal and Hoglins attack the player despite being one of them or any other mob. (attack only Skeletons).
17. Guardian's ability deals 15 points (Vanilla Guardian's damage is much less, even at high difficulty).
18. The guardian does not deal damage with spikes if he was hit by a mob or a player.
19. The skeleton does not run away from the player if he turned into a wolf. (minor error)
20. Creeper does not run away from the player if he turned into an Ocelot/Cat. (minor error)
21. Blaze does not take damage from the Snowman. (minor error)
22. Mobs that can sit on others don't bend their legs, they just stand straight.
23. When killing some baby mobs, only its adult will appear in the list.
24. The skeleton has no shooting animation.
25. Axolotl does not attack a player if that player is Drowned, Octopus or Fish.
26. Octopuses, Guardians and others have too fast swimming animation, regardless of whether the player is in place or swimming. Also Octopuses swim only in one position - horizontally.
27. Glowing Octopus does not glow.
**28**. Shulker does not stick to the block, whether on the ground, wall or ceiling. Has nothing.
29. **Wither** has no abilities, cannot fly. His heads look only towards the South.
30. **Ender Dragon** has no abilities, cannot fly. Looks only towards the North.
31. Hoglin/Zoglin does not knock up the Mob or the player on impact like Vanilla mobs do.
32. Creeper has too powerful an explosion than Vanilla Creeper. And it doesn't have an animation before the explosion.
33. The Ravager doesn't have the hit animation of the Vanilla mob.
34. Gast has no animation when shooting from his mouth.
35. Rabbit doesn't have jump animation.
36. Phantom has no ability to fly.
**37**. Zombie or Skeleton helmet runs out too fast.
38. The player, being any of the Mobs, takes damage from the Slime if he touches him.

- 

**Most important!**: **There is no list of all mobs and their varieties if the player is in Creative.** **Everyone has no sound**, whether it's walking, taking damage, or dying. For example, the passive sounds of Enderman, Skeleton or any other Mob, the splashing of fish or the Guardian if he is on land. Enderman's teleport audio is distorted. All this does not give a complete feeling that you are one of these Mobs.
All aggressive Mobs have the same damage and speed. (All this can be fixed with the "/attribute" command), (The main thing is to check the damage with _Wikipedia),_ (Carefully with the speed setting. For example, if you set the speed value like that of the Iron Golem, then when running it will be much slower than the Vanilla Mob , as well as Zombified Piglin or Wither Skeleton).

- 

_List_ _of_ _suggestions_ _for_ _Mob_ **abilities.**

1. Being a Pillager, Vex, Piglin/Brutal, pressing the ability key would raise the Mob's attacking hand, as happens with these Vanilla mobs. In parallel with this, the sounds of aggression would have long since been heard.
2. Being any type of Zombie, pressing the ability key would make the Mob's hands go up.
3. Being a large Slime/Magma Cube, smaller ones would spawn when you died.
4. As a Witch, by pressing the ability key, the player could throw potions.
5. Being a Wolf, pressing the ability key would turn his eyes red and slightly increase damage.
**6**. Vex could fly through any object (Like an observer).
7. Being a skeleton, the player could mount a Spider or a Skeleton Horse.
8. As Baby Zombie the player could mount the Chicken, as the Zombified Baby Piglin the player could mount the Baby Hoglin.
9. As a Pillager, the player could ride the Ravager.
10. Being a Silverfish, by pressing the ability key, the player could climb into stone blocks, and when pressed again, get out.
**11**. Being a Shulker, by pressing the ability key, the player could shoot at any entity, in the same way as the Guardian - by aiming the sight at the target. Also make the ability to open / close the Shulker by pressing Space. When closed, the player would have a defense of 20 units, and the arrows would be repelled, when opened, the protection would be removed.
12. Being an iron golem, by pressing the ability key, the golem would expose a hand with a flower (Like Vanilla golem).
**13**. Being an Evoker, by pressing the ability key with a cooldown of 15-20 seconds, jaws would appear (Like the Vanilla Mob), and 3 Wretches would also spawn.
14. As a Wandering Trader, pressing the ability key at night would make him invisible until the morning.
15. Being an Axolotl, by pressing the ability key, he would pretend to die and begin to regenerate (Like Vanilla mob).
16. Being a Goat, by pressing the ability key, he would deal damage from a run, and they also jump high (Like Vanilla mob).
17. Being a Polar Bear, by pressing the ability key, he would get up on his hind legs, and as a Panda, he would roll over on his back (If bamboo is in the selected slot, then when the ability is activated, Panda would sit down and eat it, instead of turning over on his back back), (I think this is difficult to implement in Mod).
18. Being a Fox, by pressing the ability key and aiming at a target, while approaching the desired distance, the player would jump to this target, and when pressing the shift, the player would curl up into a ball (Like Vanilla Mob).
19. Being a Bee, flying close to the target and pressing the ability key, the player would sting, and after a while would die, and also by pressing the right mouse button, he could fly into the Hive (Like Vanilla Mob).

**It seems that's all, I hope you read everything and it will be implemented, at least part of it. This mod will be the first in which the implementation will be so accurate!**

**Comment by FallenBO** [comment]
Of course, these are not all errors, but only visual, effects and abilities, but I tried to look.
